### PR TITLE
vendor: Update json-iterator

### PR DESCRIPTION
**Pull request by gouthamve**
Also add explicit support for NaN/Inf

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

### CODE REVIEW

File: web/api/v1/api.go [hunk 1]
```diff
@@ -1266,10 +1266,23 @@ func marshalPointJSON(ptr unsafe.Pointer, stream *jsoniter.Stream) {
 	}
 	stream.WriteMore()
 	stream.WriteRaw(`"`)
-	stream.WriteFloat64(p.V)
+
+	// Taken from https://github.com/json-iterator/go/blob/master/stream_float.go#L71 as a workaround
+	// to https://github.com/json-it
```
Comment (juliusv) [inline web/api/v1/api.go]: Was this supposed to be a link to a GitHub issue? It looks truncated.

**Comment by juliusv** [comment]
Could you elaborate on "Also add explicit support for NaN/Inf", how/where that's achieved?

**Comment by juliusv** [comment]
(just wondering because the last time I tried to update this json-iter, it resulted in NaN/Inf/-Inf causing marshaling errors, so I downgraded it again)

**Comment by juliusv** [comment]
Ah I guess that's just generally the manual `strconv.AppendFloat()`, taking that out of the hands of the now-buggy json-iter. But was there any other reason to update to the buggy json-iter version?

**Comment by bboreham** [comment]
From my point of view it's to match https://github.com/prometheus/client_golang/pull/570

However the jsoniter change was deliberate so the other choice would be to fork it?

**Comment by brian-brazil** [comment]
:+1:

File: web/api/v1/api.go [hunk 2]
```diff
@@ -1266,10 +1266,23 @@ func marshalPointJSON(ptr unsafe.Pointer, stream *jsoniter.Stream) {
 	}
 	stream.WriteMore()
 	stream.WriteRaw(`"`)
-	stream.WriteFloat64(p.V)
+
+	// Taken from https://github.com/json-iterator/go/blob/master/stream_float.go#L71 as a workaround
```
Comment (bwplotka) [inline web/api/v1/api.go]: Is this tested properly in `api_test.go`?

Comment (brian-brazil) [reply web/api/v1/api.go]: I find your lack of faith... disturbing: https://github.com/prometheus/prometheus/blob/76769d4b617989c6c60f61a3ba16261f812538c2/web/api/v1/api_test.go#L1705

Comment (bwplotka) [reply web/api/v1/api.go]: :cry: I am too lazy

**Pull request merged**